test/upgrade/preupgrade.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	pkgupgrade "knative.dev/pkg/test/upgrade"
+)
+
+// ChannelPreUpgradeTest tests channel operations before upgrade.
+func ChannelPreUpgradeTest() pkgupgrade.Operation {
+	return pkgupgrade.NewOperation("ChannelPreUpgradeTest",
+		func(c pkgupgrade.Context) {
```

Comment:
`KafkaChannelPreUpgradeTest` ?

Reply:
I wanted to have shorted names as those are already nested.

Exact test case is:  `test/upgrade: TestUpgrades/PreUpgradeTests/ChannelPreUpgradeTest/StructuredV1beta1/KafkaChannel-messaging.knative.dev/v1beta1`